modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2086,63 +2086,32 @@ def map_fn(df):
+        Reshuffle data so it would be range partitioned and then apply the passed function row-wise.
 
         Parameters
         ----------
-        axis : int or modin.core.dataframe.base.utils.Axis
-            The axis to perform the sort over.
-        columns : string or list
-            Column label(s) to use to determine lexicographical ordering.
-        ascending : boolean, default: True
-            Whether to sort in ascending or descending order.
+        key_column : hashable
+            Column name to build the range partitioning for.
+        func : callable(pandas.DataFrame) -> pandas.DataFrame
+            Function to apply against partitions.
+        ascending : bool, default: True
+            Whether the range should be built in ascending or descending order.
         **kwargs : dict
-            Keyword arguments to pass when sorting partitions.
+            Additional arguments to forward to the range builder function.
 
         Returns
         -------
-        PandasDataframe
-            A new PandasDataframe sorted into lexicographical order by the specified column(s).
+        np.ndarray
+            2D array of partitions storing result of the applied `func`.
         """
-        if not isinstance(columns, list):
-            columns = [columns]
-        # When we do a sort on the result of Series.value_counts, we don't rename the index until
-        # after everything is done, which causes an error when sorting the partitions, since the
-        # index and the column share the same name, when in actuality, the index's name should be
-        # None. This fixes the indexes name beforehand in that case, so that the sort works.
-
-        def sort_function(df):  # pragma: no cover
-            index_renaming = None
-            if any(name in df.columns for name in df.index.names):
-                index_renaming = df.index.names
-                df.index = df.index.set_names([None] * len(df.index.names))
-            df = df.sort_values(by=columns, ascending=ascending, **kwargs)
-            if index_renaming is not None:
-                df.index = df.index.set_names(index_renaming)
-            return df
-
-        axis = Axis(axis)
-        if axis != Axis.ROW_WISE:
-            raise NotImplementedError(
-                f"Algebra sort only implemented row-wise. {axis.name} sort not implemented yet!"
-            )
-
-        # If there's only one row partition can simply apply sorting row-wise without the need to reshuffle
+        # If there's only one row partition can simply apply the function row-wise without the need to reshuffle
         if self._partitions.shape[0] == 1:
-            return self.apply_full_axis(axis=1, func=sort_function)
-
-        # If this df is empty, we don't want to try and shuffle or sort.
-        if len(self.axes[0]) == 0 or len(self.axes[1]) == 0:
-            return self.copy()
+            # The method's API forces us to return raw partitions
```

Comment:
Why?

Reply:
> Do you think `_apply_func_to_range_partitioning` should return a dataframe instead of raw partitions?

Every method of the core Modin df is supposed to return a new df, whereas we return partitions from the partition manager. So I would prefer to return a df here.